src/structlog/tracebacks.py

```diff
@@ -0,0 +1,222 @@
+"""
+Extract a structured traceback from an exception.
+
+Copied and adapted from Rich:
+https://github.com/Textualize/rich/blob/972dedff/rich/traceback.py
+"""
+import os
+import sys
+
+from dataclasses import asdict, dataclass, field
+from traceback import walk_tb
+from types import TracebackType
+from typing import Any, Optional, Type, Union
+
+from .types import ExcInfo
+
+
+SHOW_LOCALS = True
+LOCALS_MAX_STRING = 80
+MAX_FRAMES = 50
+
+ExcInfo = tuple[Type[BaseException], BaseException, Optional[TracebackType]]
```

Comment:
this notation only works in annotations unfortunately. you'll have to import `Tuple`

Reply:
Yes, you are right. But it was just a prototype to demonstrate my ideas, though ;-)